src/cmd/build.rs

```diff
@@ -240,19 +250,43 @@ fn optimize_wasm(crate_metadata: &CrateMetadata) -> Result<()> {
 pub(crate) fn execute(
     manifest_path: &ManifestPath,
     verbosity: Option<Verbosity>,
-    unstable_options: UnstableFlags,
     optimize_contract: bool,
-) -> Result<PathBuf> {
-    let mut crate_metadata = CrateMetadata::collect(manifest_path)?;
-    if !optimize_contract {
-        crate_metadata.dest_wasm.set_extension("wasm.unoptimized");
+    skip_bundle: bool,
+    skip_metadata: bool,
```

Comment:
I think `false, true` here is an invalid combination (bundling has to create the metadata anyway). 

To make it more readable and prevent invalid states I'd prefer an enum here rather than multiple bool flags e.g.

```
enum BuildArtifacts {
  All, // false, false
  CodeOnly, // true, true
  MetadataOnly, // false, true
}

impl BuildArtifacts {
  fn from_flags(..) -> Result<BuildArtifacts> { ..}
```

Also, at the risk of reopening the bikeshedding, we could consider using equivalent command options e.g. `--bundle` (default) `--code-only` and `--metadata-only`. :see_no_evil: